### DISCUSSION TITLE
Ability to skip install of KB892130 on Windows XP

### DISCUSSION THREAD

**Discussion by atauenis** [LegacyUpdate/LegacyUpdate#146] [Ideas]
After installing of Legacy Update 1.6.2 on Windows XP Professional SP1a, it required to install 4 components: WinXP Update KB898461, BITS & WinHTTP update KB842773, MSI 3.1 (no KB nr.) and WGA (KB892130). Instead I've waited a minute, and WU Client pops up with invite to install only 3 really important updates. 

![2023-05-08_22-49-04](https://github.com/kirb/LegacyUpdate/assets/4800266/162bf37b-d0b2-4308-995f-5772e683e2cb)
Okay, installed them. Good news: LU is correctly working with English WinXP localized via MUI.

Then the WU Client found the rest of XP SP1 updates plus SP2 + WGA KB892130 (approx. 60 updates total). Removed check from SP2 and WGA, and all ~58 updates quickly got successfully installed.
![2023-05-08_22-53-21](https://github.com/kirb/LegacyUpdate/assets/4800266/e507e50c-5f39-473b-840e-878d4d0fe6a4)

However, LegacyUpdate.net still wants KB892130 (WGA). And WU Client still don't show any optional updates and updates for other installed software (Office 2002, Visual Studio 2003). Only Update History is displaying correctly.
![LegacyUpdate KB892130](https://github.com/kirb/LegacyUpdate/assets/4800266/8ed66f58-abc0-4d2a-bd7e-d285d62ed6f1)

The OS was installed from a real MSDN Subscription DVD-ROM, but install keys were lost (with access to the MSDN account used in 2004 to order the disk), and a googled Retail key were used to install (plus some registry tweak to shut down the WPA).
![Untitled FR12](https://github.com/kirb/LegacyUpdate/assets/4800266/d6cf302d-1306-4866-9fbc-d4dacf492413)

Is it possible to bypass KB892130 installation in order to made LegacyUpdate fully working?

**Comment by atauenis** [comment]
Waited a few more time, and updates for Office XP & Visio 2002 was found and successfully installed. So the real problem is only with web site part. In fact, Windows Update Agent is perfectly working without both XP SP2 and WGA.

**Reply by kirb** [reply]
Jealous that you have your own original MSDN discs 😁 (I’m surprised they bundled XP RTM builds alongside SP1a. Did they remove the Microsoft JVM from them?)

I’ve been uneasy about changing how the WGA forced install works, actually I’m very surprised nobody else has ever brought it up before. I *did* remove the WGA validation on the website, because I’d have to reimplement the ActiveX control (LegitCheckControl.dll) that handles it - it just seems less risky to the project to remove it than to try and reimplement their DRM. The logic to do with Windows Update hiding access to optional updates on non-genuine installs is fully implemented inside the website. That said, I don’t know if the Automatic Updates dialog still has this behavior. It’s possible it doesn’t.

I can probably have the update ignored by simply adding an `AND id != 'whatever the guid is'` to the update searching query. Need to test that this won’t cause something to break e.g. any updates that themselves check for WGA Notification being installed, for any version of XP, Office 2002/2003, and whatever else.

**Reply by atauenis** [reply]
>(I’m surprised they bundled XP RTM builds alongside SP1a. Did they remove the Microsoft JVM from them?)

Yes, XP Pro RTM copy on CD seems to be JVM-less. Just like SP1a.
![XP RTM MSDN no JVM](https://github.com/kirb/LegacyUpdate/assets/4800266/c9a2d1bc-b7ab-4469-a360-a9663027030f)

>Need to test that this won’t cause something to break e.g. any updates that themselves check for WGA Notification being installed, for any version of XP, Office 2002/2003, and whatever else.

AFAIR, only IE7, WMP11 and some other large updates rely on WGA. Probably anything released before WGA creation (it was released near 2006), would not check system legality. Some MS Office updates checks serial number of MS Office installation, but not Windows serial number.

**Reply by sal-ammoniac** [reply]
try to put LegitCheckControl.dll under %windir%/system32 and register it?

**Reply by atauenis** [reply]
@sal-ammoniac , doesn't help (used v1.3.272.0 from KB938720 article).

**Comment by kristibektashi** [comment]
Even if the logic for blocking other updates when WGA isn't installed is removed from the website it should still be optionally installable (with the option to hide it of course) so that it can still be installed in case some app requires it

**Comment by 760ceb3b9c0ba4872cadf3ce35a7a494** [comment]
hi! I managed to skip KB892130, i wrote the steps down here: https://760ceb3b9c0ba4872cadf3ce35a7a494.neocities.org/posts/skip-KB892130-on-windows-xp/
